src/middleware/getInfinityMsg/getInfinityLocalMsg.middleware.tsx

```diff
@@ -0,0 +1,164 @@
+
+		if (assistanceDice && assistanceDiceResults?.length) {
+			const assistanceDiceResultsJoined = joinAsBlocks(assistanceDiceResults);
+			const assistanceComplications = assistanceSuccessLevel.complications
+				? <p className={styles.assistanceResultRow}>Complications: <CodeSpan type="failure">{assistanceSuccessLevel.complications}</CodeSpan></p>
+				: null;
+
+			fields.push(
+				<div className={classNames([styles.infinityResultDetails, styles.infinityResultDetailsAssistance])}>
+					<p className={styles.resultDetailsRow}><strong>Assistance Roll:</strong></p>
+					<p className={styles.resultDetailsRow}>Rolled: {assistanceDiceResultsJoined}</p>
+					{ assistanceFocus !== '' && <p className={styles.resultDetailsRow}>Assistance Focus: <CodeSpan>{assistanceFocus}</CodeSpan></p> }
+					{ assistanceTn !== '' && <p className={styles.resultDetailsRow}>Assistance TN: <CodeSpan>{assistanceTn}</CodeSpan></p> }
+					{ assistanceUntrainedTest && <p className={styles.resultDetailsRow}>Assistance Untrained Test</p> }
+					<p className={styles.resultDetailsRow}>Successes: <CodeSpan type={assistanceSuccessLevel.successLevel > 0 ? 'success' : ''}>{assistanceSuccessLevel.successLevel}</CodeSpan></p>
+					{ assistanceComplications }
+					<InfoTooltip content={tooltip.assistance} className={styles.assistanceIcon} />
+				</div>
+			);
+		}
+	
+		const labels: labelsType = {
+			result: 'Successes',
+			vs: 'Difficulty'
+		};
+	
+		fields.push(
+			<ResultVsSkillRow
+				skillLevel={difficulty}
+				finalDieResult={successLevel.successLevel}
+				isSuccess={successLevel.isSuccess}
+				labels={labels}
+			/>
+		);
+	
+		if (successLevel.isSuccess) {
+			fields.push(
+				<div className={cx({ generalResult: true, generalResultSuccess: true })}>Success</div>
+			);
+		} else {
+			fields.push(
+				<div className={cx({ generalResult: true, generalResultFailure : true })}>Failure</div>
+			);
+		}
+	
+		fields.push(
+			<div className={cx({ slResult: true, momentumResults: true })}>
+				<div>
+					{/* COMPLICATIONS */}
+					<div>
+						<span className={cx({ slResultLabel: true, inactive: successLevel.complications === 0 })}>Complications:</span>
+						</div>
+					<div>
+						<CodeSpan
+							className={styles.slResultSpan}
+							type ={successLevel.complications > 0 ? 'failure' : 'inactive'}
+					>{successLevel.complications}</CodeSpan>
+						</div>
+				</div>
+				<div>
+					{/* MOMENTUM */}
+					<div>
+						<span className={cx({ slResultLabel: true, inactive: successLevel.momentum === 0 })}>Momentum:</span>
+						</div>
+					<div>
+						<CodeSpan
+							className={styles.slResultSpan}
+							type={successLevel.momentum > 0 ? 'success' : 'inactive'}
+						>{successLevel.momentum}</CodeSpan>
+					</div>
+				</div>
+	
+			</div>
+		);	
+
+		store.dispatch(localMsgReady({
+			title,
+			fields,
+			isSuccess: successLevel.isSuccess,
+			isFailure: successLevel.isFailure,
+			rollOptions,
+			results
+		}));
+	}
+	next(action);
+};
```

Comment:
I understand why you made it this way and I think it's fine for now. However, take a look at games that were added recently - esp. Legend of the Five Rings. I use new approach there instead of pushing elements into final results container. The new approach uses separate modal for showing results, so only data has to be provided. If you consider adding any more games, please use the newer approach next time.

Reply:
I come from Angular, so I think I follow what you mean.  And I agree, will do!